FirebaseSessions/Sources/SettingsDownloadClient.swift

```diff
@@ -0,0 +1,99 @@
+//
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import Foundation
+
+@_implementationOnly import GoogleUtilities
+
+protocol SettingsDownloadClient {
+  func fetch(completion: @escaping (Result<[String: Any], Error>) -> Void)
+}
+
+class SettingsDownloader: SettingsDownloadClient {
+  private let appInfo: ApplicationInfoProtocol
+  private let identifiers: IdentifierProvider
+  private let installations: InstallationsProtocol
+
+  init(appInfo: ApplicationInfoProtocol, identifiers: IdentifierProvider,
+       installations: InstallationsProtocol) {
+    self.appInfo = appInfo
+    self.identifiers = identifiers
+    self.installations = installations
+  }
+
+  func fetch(completion: @escaping (Result<[String: Any], Error>) -> Void) {
+    guard let url = url else {
+      completion(.failure(FirebaseSessionsError.SettingsError("Invalid URL")))
+      return
+    }
+
+    buildRequest(url: url) { result in
+      switch result {
+      case let .success(request):
+        let task = URLSession.shared.dataTask(with: request) { data, response, error in
+          if let data = data {
+            if let dict = try? JSONSerialization.jsonObject(with: data) as? [String: Any] {
+              completion(.success(dict))
+            } else {
+              completion(.failure(FirebaseSessionsError
+                  .SettingsError("Failed to parse JSON to dictionary")))
+            }
+          } else if let error = error {
+            completion(.failure(FirebaseSessionsError.SettingsError(error.localizedDescription)))
+          }
+        }
+        task.resume()
```

Comment:
I'm curious what this line is for. Could we add a comment too?

Reply:
Ah I understand now: https://developer.apple.com/documentation/foundation/nsurlsessiontask/1411121-resume - this seems to just start the download

If we could add a comment that might help our future selves

Reply:
Done